src/app/AttributePathExpandIterator.cpp

```diff
@@ -162,14 +156,28 @@ std::optional<AttributeId> AttributePathExpandIterator::NextAttributeId()
             //
             // For wildcard expansion, we validate that this is a valid attribute for the given
             // cluster on the given endpoint. If not a wildcard expansion, return it as-is.
-            if (mPosition.mAttributePath->mValue.IsWildcardPath())
+            DataModel::AttributeFinder finder(mDataModelProvider);
+
+            const ConcreteAttributePath attributePath(mPosition.mOutputPath.mEndpointId, mPosition.mOutputPath.mClusterId,
+                                                      mPosition.mAttributePath->mValue.mAttributeId);
+            std::optional<DataModel::AttributeEntry> entry = finder.Find(attributePath);
+
+            if (entry.has_value() || mPosition.mAttributePath->mValue.IsWildcardPath())
             {
-                if (!IsValidAttributeId(mPosition.mAttributePath->mValue.mAttributeId))
-                {
-                    return std::nullopt;
-                }
+                return entry;
             }
-            return mPosition.mAttributePath->mValue.mAttributeId;
+
+            // We get here if:
+            //   - entry is NOT valid (this is not a valid attribute)
+            //   - path is NOT a wildcard (i.e. we were asked to explicitly return it)
+            // as a result, we have no way to generate a "REAL" attribute metadata, so return one
+            // that is not accessible at all
+            return DataModel::AttributeEntry{
```

Comment:
Seems to me like in this case the _entry_ we give people better be nullopt to indicate the truth of the world: there is no entry.  But we do want to treat this case as having a path, unlike the wildcard case, right?

Reply:
Updated to reset the entry in this case, so it is always reflecting real "what is the entry" data.